src/api/component.js

```diff
@@ -90,6 +91,10 @@ function createInitProps (Ctor) {
   };
 }
 
+function tagName (elem) {
+  return elem && elem.tagName ? elem.tagName.toLowerCase() : '';
```

Comment:
Should be able to rely on `elem` to always be there. You can also use `localName` instead of `tagName.toLowerCase()`.